### PR TITLE
[14.0][IMP] product_seasonality: Add product template on config lines

### PR DESCRIPTION
**Pull request by jcoux**

### CODE REVIEW

File: product_seasonality/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {
     "name": "Product seasonality",
     "summary": "Define rules for products' seasonal availability",
-    "version": "14.0.1.0.1",
+    "version": "14.0.1.0.2",
```
Comment (simahawk) [inline product_seasonality/__manifest__.py]: this is not a fix but a feature

File: product_seasonality/migrations/14.0.1.0.2/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    if not version:
+        return
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    _logger.info("Set product_template_id on seasonal config lines")
+    config_lines = env["seasonal.config.line"].search([])
```
Comment (simahawk) [inline product_seasonality/migrations/14.0.1.0.2/post-migration.py]:
```suggestion
    config_lines = env["seasonal.config.line"].search([("product_template_id", "=", False)])
```

File: product_seasonality/models/seasonal_config_line.py [hunk 3]
```diff
@@ -54,7 +58,11 @@ def is_sale_ok(self, date):
 
     def find_for_product(self, prod, config=None):
         domain = [
+            "|",
             ("product_id", "=", prod.id),
+            "&",
+            ("product_id", "=", False),
+            ("product_template_id", "=", prod.product_tmpl_id.id),
```
Comment (simahawk) [inline product_seasonality/models/seasonal_config_line.py:65]: add a test pls

**Review by simahawk — CHANGES_REQUESTED** [review]

File: product_seasonality/models/seasonal_config_line.py (hunk 3 above)
Comment (jcoux) [reply product_seasonality/models/seasonal_config_line.py:65]: Test already done by this little change:
https://github.com/OCA/product-attribute/pull/871/files#diff-ca80e35cf3f4f78c374cbae4e772aea98dad0bda75d76628eb3166dbe9f5503fR52

With it we have a line with product variant, and one without.

File: product_seasonality/__manifest__.py (hunk 1 above)
Comment (jcoux) [reply product_seasonality/__manifest__.py]: Fixed with commit amend/push force: https://github.com/OCA/product-attribute/compare/0fe21452c8a4ee85a1b20e99421a67a191ec13b2..86df0a88738a1159fdebfa1ba3dd9eed9d226cd5

File: product_seasonality/migrations/14.0.1.0.2/post-migration.py (hunk 2 above)
Comment (jcoux) [reply product_seasonality/migrations/14.0.1.0.2/post-migration.py]: Fixed with commit amend/push force: https://github.com/OCA/product-attribute/compare/0fe21452c8a4ee85a1b20e99421a67a191ec13b2..86df0a88738a1159fdebfa1ba3dd9eed9d226cd5

**Review by simahawk — APPROVED** [review]

**Review by damdam-s — APPROVED** [review]

**Pull request merged**